http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieRuntimeMonitor.scala

```diff
@@ -0,0 +1,169 @@
+              case true =>
+                // If the project is active, check each individual cluster
+                clusters.toList.traverseFilter { cluster =>
+                  isRuntimeActiveInGoogle(cluster, startInstant).ifA(F.pure(None), F.pure(Some(cluster)))
+                }
+              case false =>
+                // If the project is inactive, all clusters in the project are zombies
+                F.pure(clusters.toList)
+            }
+          )
+      }
+      // Error out each detected zombie cluster
+      _ <- zombies.parTraverse(zombie => semaphore.withPermit(handleZombieCluster(zombie, startInstant)))
+      end <- Timer[F].clock.realTime(TimeUnit.MILLISECONDS)
+      duration = end - start
+      _ <- logger.info(
+        s"Detected ${zombies.size} zombie clusters in ${zombies.map(_.googleProject).toSet.size} projects. Elapsed time = ${duration} milli seconds"
+      )
+    } yield ()
+
+  private def isProjectActiveInGoogle(googleProject: GoogleProject): F[Boolean] = {
+    // Check the project and its billing info
+    val res = for {
+      isBillingActive <- F.liftIO(IO.fromFuture(IO(googleProjectDAO.isBillingActive(googleProject.value))))
+      // short circuit
+      isProjectActive <- if (!isBillingActive) F.pure(false)
+      else F.liftIO(IO.fromFuture(IO(googleProjectDAO.isProjectActive(googleProject.value))))
+    } yield isProjectActive
+
+    res.recoverWith {
+      case e: GoogleJsonResponseException if e.getStatusCode == 403 =>
+        logger
+          .info(e)(
+            s"Unable to check status of project ${googleProject.value} for zombie cluster detection " +
+              s"due to a 403 from google. We are assuming this is a free credits project that has been cleaned up. " +
+              s"Marking project as a zombie."
+          )
+          .as(false)
+
+      case e =>
+        logger
+          .warn(e)(s"Unable to check status of project ${googleProject.value} for zombie cluster detection")
+          .as(true)
+    }
+  }
+
+  private def isRuntimeActiveInGoogle(runtime: PotentialZombieRuntime, now: Instant)(implicit traceId: ApplicativeAsk[F, TraceId]): F[Boolean] = {
+    val milliSecondsSinceClusterCreation: Long = now.toEpochMilli - runtime.auditInfo.createdDate.toEpochMilli
+    // this or'd with the google cluster status gives creating clusters a grace period before they are marked as zombies
+    if (runtime.status == RuntimeStatus.Creating && milliSecondsSinceClusterCreation < config.creationHangTolerance.toMillis) {
+      F.pure(true)
+    } else {
+      val runtimeStatus: F[RuntimeStatus] = runtime.cloudService match {
+        case CloudService.GCE =>
+          gce.getInstance(runtime.googleProject, config.gceZoneName, InstanceName(runtime.runtimeName.asString)).map{
+            instance =>
+              instance.flatMap(s => RuntimeStatus.withNameInsensitiveOption(s.getStatus)).getOrElse(RuntimeStatus.Unknown)
```

Comment:
Some GCE statuses aren't captured in `RuntimeStatus`. So this will resolve to `Unknown` but I think that should work fine since `Unknown` is in `activeStatuses`..

Reply:
I actually run into this in unit tests...added all gce statuses